BSP/Simulator/Src/BSP_Precharge.c

```diff
@@ -14,43 +14,15 @@
 */
 void BSP_Precharge_Write(board_t board, State status){
     // first get previous board states from csv file
-    FILE* fp = fopen(FILE_NAME, "r");
-
-    if(!fp){
-        printf("Precharge file is not found\n");
-        return;
-    }
-
-    int fno = fileno(fp);
-    flock(fno,LOCK_EX);
-
-    int data = 0;
-    fscanf(fp,"%d", &data);
+    int data = BSP_GPIO_Read();
```

Comment:
You'll end up passing in a parameter here. It'll probably be an enum, so don't worry about it for now. Just a heads up